app/src/main/java/bg/zahov/app/ui/home/HomeViewModel.kt

```diff
@@ -25,45 +29,56 @@ import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 import java.time.temporal.TemporalAdjusters
 
+data class HomeUiState(
+    val isLoading: Boolean = false,
+    val username: String = "",
+    val numberOfWorkouts: String = "",
+    val barData: BarData = BarData(),
+    val isChartLoading: Boolean = true
+)
+
+data class BarData(
+    var chartVisibility: Int = View.GONE,
+    var xMin: Float = 0f,
+    var xMax: Float = 0f,
+    var yMin: Float = 0f,
+    var yMax: Float = 0f,
+    var chartData: List<BarEntry> = listOf(),
+    var weekRanges: List<String> = listOf(),
+    var xValueFormatter: ValueFormatter = IndexAxisValueFormatter(weekRanges.toTypedArray())
```

Comment:
This doesn't seem like the place to set and reuse a formatter. Please move it in the VM/View